src/Umbraco.Web/Security/Identity/AppBuilderExtensions.cs

```diff
@@ -213,15 +234,81 @@ public static IAppBuilder UseUmbracoBackOfficeExternalCookieAuthentication(this
                 CookieSecure = GlobalSettings.UseSSL ? CookieSecureOption.Always : CookieSecureOption.SameAsRequest,
                 CookieHttpOnly = true,
                 CookieDomain = UmbracoConfig.For.UmbracoSettings().Security.AuthCookieDomain
-            });
+            }, stage);
+
+            return app;
+        }
+
+        /// <summary>
+        /// In order for preview to work this needs to be called
+        /// </summary>
+        /// <param name="app"></param>
+        /// <param name="appContext"></param>
+        /// <returns></returns>
+        /// <remarks>
+        /// This ensures that during a preview request that the back office use is also Authenticated and that the back office Identity
+        /// is added as a secondary identity to the current IPrincipal so it can be used to Authorize the previewed document.
+        /// </remarks>
+        /// <remarks>
+        /// By default this will be configured to execute on PipelineStage.PostAuthenticate
+        /// </remarks>
+        public static IAppBuilder UseUmbracoPreviewAuthentication(this IAppBuilder app, ApplicationContext appContext)
+        {
+            return app.UseUmbracoPreviewAuthentication(appContext, PipelineStage.PostAuthenticate);
+        }
+
+        /// <summary>
+        /// In order for preview to work this needs to be called
+        /// </summary>
+        /// <param name="app"></param>
+        /// <param name="appContext"></param>
+        /// <param name="stage"></param>
+        /// <returns></returns>
+        /// <remarks>
+        /// This ensures that during a preview request that the back office use is also Authenticated and that the back office Identity
+        /// is added as a secondary identity to the current IPrincipal so it can be used to Authorize the previewed document.
+        /// </remarks>
+        public static IAppBuilder UseUmbracoPreviewAuthentication(this IAppBuilder app, ApplicationContext appContext, PipelineStage stage)
+        {
+            //don't apply if app isnot ready
+            if (appContext.IsUpgrading || appContext.IsConfigured)
```

Comment:
Copy paste fail. Some other middleware needs to be enabled during an upgrade so that user can be auth'd during the installation process... but for preview you are 100% correct it shouldn't be enabled during upgrade.